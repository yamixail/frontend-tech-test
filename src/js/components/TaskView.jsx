import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class TaskView extends PureComponent {
  render() {
    const {
      id, title, description, activateEditMode, deleteTask,
    } = this.props;

    return (
      <div className="task">
        <div className="task__title">
          <button type="button" title="delete task" className="task__delete" onClick={deleteTask}>&times;</button>
          <button type="button" title="edit task" className="task__edit" onClick={activateEditMode}>&#9998;</button>
          {`${id}. ${title}`}
        </div>
        <div className="task__description">
          {description}
        </div>
      </div>
    );
  }
}

TaskView.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  activateEditMode: PropTypes.func.isRequired,
  deleteTask: PropTypes.func.isRequired,
};

export default TaskView;
