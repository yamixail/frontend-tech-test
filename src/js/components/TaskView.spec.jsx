import React from 'react';
import { shallow } from 'enzyme';

import TaskView from './TaskView';

const noop = () => {};

describe('TaskView', () => {
  it('should render', () => {
    const wrapper = shallow(
      <TaskView
        id={1}
        title="Title"
        description="Lorem ipsum dolor sit amet"
        activateEditMode={noop}
        deleteTask={noop}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
