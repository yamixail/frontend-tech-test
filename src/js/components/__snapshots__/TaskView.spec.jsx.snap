// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView should render 1`] = `
<div
  className="task"
>
  <div
    className="task__title"
  >
    <button
      className="task__delete"
      onClick={[Function]}
      title="delete task"
      type="button"
    >
      ×
    </button>
    <button
      className="task__edit"
      onClick={[Function]}
      title="edit task"
      type="button"
    >
      ✎
    </button>
    1. Title
  </div>
  <div
    className="task__description"
  >
    Lorem ipsum dolor sit amet
  </div>
</div>
`;
